ocs_ci/helpers/mcg_stress_helper.py

```diff
@@ -0,0 +1,618 @@
+        f"DELETE OP: Delete objects recursively from the bucket "
+        f"{bucket_name} under prefix {prev_iteration}"
+    )
+
+    rm_object_recursive(
+        pod_obj,
+        bucket_name,
+        mcg_obj,
+        prefix=prev_iteration,
+        timeout=timeout,
+    )
+    logger.info(
+        f"DELETE OP: Total of {(multiplier-1)*1000000} objects got deleted "
+        f"in the current iteration {current_iteration}"
+    )
+    logger.info(
+        f"DELETE OP: Successfully completed object deletion operation on "
+        f"bucket {bucket_name} under prefix {prev_iteration}"
+    )
+
+
+def list_objs_from_bucket(bucket, prev_iteration, event=None):
+    """
+    List objects from bucket
+
+    Args:
+        bucket (Tuple): Tuple consisting of backend storage type and bucket object
+        prev_iteration (int): Iteration number or prefix from where should list objects
+        event (threading.Event()): Event object to signal the execution
+            completion
+
+    """
+    current_iteration = prev_iteration + 1
+    bucket_type, bucket_obj = bucket
+    bucket_name = bucket_obj.name
+    if bucket_type.upper() == "RGW":
+        mcg_obj = OBC(bucket_name)
+    else:
+        mcg_obj = MCG()
+
+    logger.info(
+        f"LIST OP: Listing objects from the bucket {bucket_name} "
+        f"under prefix {prev_iteration}"
+    )
+    while True:
+        s3_list_objects_v2(
+            mcg_obj, bucket_name, prefix=str(prev_iteration), delimiter="/"
+        )
+
+        if event.is_set():
+            logger.info(
+                f"LIST OP: Total of {current_iteration*1000000} objects got listed "
+                f"in the current iteration {current_iteration}"
+            )
+            logger.info(
+                f"LIST OP: Successfully completed object list operation on "
+                f"bucket {bucket_name} under prefix {prev_iteration}"
+            )
+            break
+
+
+def download_objs_from_bucket(
+    pod_obj, bucket, target_dir, prev_iteration, event=None, multiplier=1
+):
+    """
+    Download objects from a bucket back to local directory
+
+    Args:
+        pod_obj (Pod): Noobaa stress CLI pod object
+        bucket (Tuple): Tuple consisting of backend storage type and bucket object
+        target_dir (str): Target directory to download objects
+        prev_iteration (int): Iteration number or prefix from where should download objects
+        event (threading.Event()): Event object to signal the execution
+            completion
+
+    """
+    current_iteration = prev_iteration + 1
+    bucket_type, bucket_obj = bucket
+    bucket_name = bucket_obj.name
+    base_timeout = 20000
+    timeout = base_timeout * multiplier
+
+    if bucket_type.upper() == "RGW":
+        mcg_obj = OBC(bucket_name)
+    else:
+        mcg_obj = MCG()
+
+    logger.info(
+        f"DOWNLOAD OP: Download objects from the bucket "
+        f"{bucket_name} under prefix {prev_iteration}"
+    )
+    while True:
```

Comment:
while True loop is being used, which can be inefficient and may lead to endless loops if the condition to exit isn't properly met. Is there any way, we can replace it? something like setting timeout, limiting the iter or chuncking etc?

Reply:
@PrasadDesala will address this in next PR if there is any better way.